services/QuillLMS/app/services/open_ai/save_translated_text.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module OpenAI
+  class SaveTranslatedText < ApplicationService
+
+    class OpenAITranslationError < StandardError; end
+    attr_accessor :english_text
+
+    def initialize(english_text)
```

Comment:
I would pull locale up to the initializer that way you are making the dependency more visible:
````
def initialize(english_text:, locale: Gengo::SPANISH_LOCALE)
  @english_text = english_text
  @locale = locale
end

# Yielding a simpler construction below
private def translated_text = @translated_text ||= TranslatedText.find_or_initialize_by(english_text:, locale:)
```